### PR TITLE
Add backend server

### PR DESCRIPTION
**Pull request by HeinrichvonStein**
These changes add a backend API to the demo app. 

### Future improvements
- Add AuthContext

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,7 +54,7 @@ This app focuses on showing how to use PowerSync to sync items and view them off
 From the project root directory, run:
 
 ```sh
-npm install
+tnpm install
```
Comment (michaelbarnes) [inline README.md]: @HeinrichvonStein Seems there is a typo here

File: README.md [hunk 2]
```diff
@@ -52,16 +63,24 @@ Copy the `.env.template` file and update the variables accordingly:
...
 ```
 
-### Run the App
+## Run the App
+
+Before running the app, start the backend server:
 
+```bash
+cd backend
+npm start
```
Comment (michaelbarnes) [inline README.md:74]: @HeinrichvonStein Please add a `npm install` command above this line.

File: backend/package.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "attendee-list-backend",
```
Comment (michaelbarnes) [inline backend/package.json]: Can you change the package name, please?

File: backend/tsconfig.json
Comment (michaelbarnes) [inline backend/tsconfig.json:1]: Can you remove all of the commented out options for the compiler, please?

**Review by michaelbarnes — CHANGES_REQUESTED** [review]

File: README.md
Comment (michaelbarnes) [inline README.md:1]: Can we add a small section in the README.md exaplaining that a backend component has been added to the app and that it uses express, Node.js and TypeScript?

Comment (HeinrichvonStein) [reply README.md:1]: @michaelbarnes I mentioned that in the [beginning](https://github.com/powersync-ja/powersync-react-native-mongodb-mbnb/pull/1/commits/5e50432fccd4345c815598bf555e64815e875192#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R6-R7) of the README. I can extend it if you'd like?

File: README.md [hunk 4]
```diff
@@ -3,6 +3,8 @@
 This is a demo application which was used to port a Realm based application to PowerSync. The original Realm
 implementation can be [found here](https://github.com/realm/realm-js/tree/main/examples/rn-mbnb).
 
+This repo also contains a Node.js backend API that receives upload from the client.
```
Comment (michaelbarnes) [inline README.md]: This repo also contains a Node.js backend API using Express and TypeScript. This API will receive the changes uploaded by the client and will write them back to the MongoDB database. You can run this alongside the frontend project.

File: README.md (hunk 2 above)
Comment (michaelbarnes) [reply README.md:74]: Swap the cd command and npm install around.

File: backend/package.json (hunk 3 above)
Comment (michaelbarnes) [reply backend/package.json]: Change the pnpm references to npm

**Review by michaelbarnes — APPROVED** [review]
LGTM

**Pull request merged**